### ISSUE TITLE
CRAN build failing on linux

### ISSUE THREAD

**Issue by mem48** [ropensci/opentripplanner#34]
@Robinlovelace can you confrim that you are getting these errors on Linux?

https://win-builder.r-project.org/incoming_pretest/opentripplanner_0.2.0.0000_20191129_213949/Debian/00check.log

It builds fine on windows, and the CI is also fine.

**Comment by Robinlovelace** [comment]
I'll give it a spin.

**Comment by Robinlovelace** [comment]
Seems to work on my main Ubuntu 18.04 machine:

``` r
remotes::install_github("ropensci/opentripplanner", force = T); library(opentripplanner); Sys.info()
#> Downloading GitHub repo ropensci/opentripplanner@master
#> 
#>   
   checking for file ‘/tmp/Rtmp99qFsD/remotes71205d599cfa/ropensci-opentripplanner-2535fc8/DESCRIPTION’ ...
  
✔  checking for file ‘/tmp/Rtmp99qFsD/remotes71205d599cfa/ropensci-opentripplanner-2535fc8/DESCRIPTION’
#> 
  
─  preparing ‘opentripplanner’:
#>    checking DESCRIPTION meta-information ...
  
✔  checking DESCRIPTION meta-information
#> 
  
─  checking for LF line-endings in source and make files and shell scripts
#> 
  
─  checking for empty or unneeded directories
#> 
  
─  building ‘opentripplanner_0.2.0.0000.tar.gz’
#> 
  
   
#> 
#> Installing package into '/home/robin/R/x86_64-pc-linux-gnu-library/3.6'
#> (as 'lib' is unspecified)
#>                                       sysname 
#>                                       "Linux" 
#>                                       release 
#>                           "4.15.0-70-generic" 
#>                                       version 
#> "#79-Ubuntu SMP Tue Nov 12 10:36:11 UTC 2019" 
#>                                      nodename 
#>                              "robin-Poseidon" 
#>                                       machine 
#>                                      "x86_64" 
#>                                         login 
#>                                     "unknown" 
#>                                          user 
#>                                       "robin" 
#>                                effective_user 
#>                                       "robin"
```

<sup>Created on 2019-11-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.6.1 (2019-07-05)
#>  os       Ubuntu 18.04.3 LTS          
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language en_GB:en                    
#>  collate  en_GB.UTF-8                 
#>  ctype    en_GB.UTF-8                 
#>  tz       Europe/London               
#>  date     2019-11-29                  
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package         * version    date       lib
#>  assertthat        0.2.1      2019-03-21 [2]
#>  backports         1.1.5      2019-10-02 [1]
#>  callr             3.3.2      2019-09-22 [1]
#>  cli               1.1.0      2019-03-19 [2]
#>  crayon            1.3.4      2017-09-16 [2]
#>  curl              4.2        2019-09-24 [1]
#>  desc              1.2.0      2018-05-01 [2]
#>  devtools          2.2.1      2019-09-24 [1]
#>  digest            0.6.23     2019-11-23 [1]
#>  ellipsis          0.3.0      2019-09-20 [1]
#>  evaluate          0.14       2019-05-28 [2]
#>  fs                1.3.1      2019-05-06 [2]
#>  glue              1.3.1      2019-03-12 [2]
#>  highr             0.8        2019-03-20 [3]
#>  htmltools         0.4.0      2019-10-04 [3]
#>  knitr             1.26       2019-11-12 [1]
#>  magrittr          1.5        2014-11-22 [2]
#>  memoise           1.1.0      2017-04-21 [3]
#>  opentripplanner * 0.2.0.0000 2019-11-29 [1]
#>  pkgbuild          1.0.6      2019-10-09 [2]
#>  pkgload           1.0.2      2018-10-29 [3]
#>  prettyunits       1.0.2      2015-07-13 [2]
#>  processx          3.4.1      2019-07-18 [2]
#>  ps                1.3.0      2018-12-21 [2]
#>  R6                2.4.1      2019-11-12 [1]
#>  Rcpp              1.0.3      2019-11-08 [1]
#>  remotes           2.1.0      2019-06-24 [1]
#>  rlang             0.4.2      2019-11-23 [1]
#>  rmarkdown         1.18       2019-11-27 [1]
#>  rprojroot         1.3-2      2018-01-03 [2]
#>  sessioninfo       1.1.1      2018-11-05 [3]
#>  stringi           1.4.3      2019-03-12 [2]
#>  stringr           1.4.0      2019-02-10 [2]
#>  testthat          2.3.0      2019-11-05 [2]
#>  usethis           1.5.1      2019-07-04 [1]
#>  withr             2.1.2      2018-03-15 [2]
#>  xfun              0.11       2019-11-12 [1]
#>  yaml              2.2.0      2018-07-25 [3]
#>  source                                   
#>  CRAN (R 3.6.0)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.5.3)                           
#>  standard (@1.3.4)                        
#>  CRAN (R 3.6.1)                           
#>  standard (@1.2.0)                        
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.0)                           
#>  CRAN (R 3.6.0)                           
#>  CRAN (R 3.5.3)                           
#>  CRAN (R 3.5.3)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.5.2)                           
#>  CRAN (R 3.5.0)                           
#>  Github (ropensci/opentripplanner@2535fc8)
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.5.1)                           
#>  CRAN (R 3.5.3)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.5.3)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.5.3)                           
#>  CRAN (R 3.5.1)                           
#>  CRAN (R 3.5.3)                           
#>  standard (@1.4.0)                        
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.5.3)                           
#>  CRAN (R 3.6.1)                           
#>  CRAN (R 3.5.1)                           
#> 
#> [1] /home/robin/R/x86_64-pc-linux-gnu-library/3.6
#> [2] /usr/local/lib/R/site-library
#> [3] /usr/lib/R/site-library
#> [4] /usr/lib/R/library
```

</details>

**Comment by Robinlovelace** [comment]
Update: this is not OS specific. Expect a PR fixing this soon @mem48.

**Comment by mem48** [comment]
Any suggestions on how to fix this? I suppose I need a VM with Debian

**Comment by mem48** [comment]
Thanks

**Comment by Robinlovelace** [comment]
Give it a spin here if you want ; ) https://rstudio.robinlovelace.net/

**Comment by Robinlovelace** [comment]
There are a number of tests that fail. Does `devtools::check()` pass for you?

**Comment by Robinlovelace** [comment]
I get this:

``` r
devtools::check("~/wip/opentripplanner/")
#> Updating opentripplanner documentation
#> Writing NAMESPACE
#> Loading opentripplanner
#> Writing NAMESPACE
#> ── Building ──────────────────────────────────────────────────────────────────────────────────────────────────────── opentripplanner ──
#> Setting env vars:
#> ● CFLAGS    : -Wall -pedantic
#> ● CXXFLAGS  : -Wall -pedantic
#> ● CXX11FLAGS: -Wall -pedantic
#> ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
#>   
   checking for file ‘/home/robin/wip/opentripplanner/DESCRIPTION’ ...
  
✔  checking for file ‘/home/robin/wip/opentripplanner/DESCRIPTION’
#> 
  
─  preparing ‘opentripplanner’:
#>    checking DESCRIPTION meta-information ...
  
✔  checking DESCRIPTION meta-information
#> 
  
─  installing the package to build vignettes
#> 
  
   creating vignettes ...
  
✔  creating vignettes (2.1s)
#> 
  
─  checking for LF line-endings in source and make files and shell scripts
#> 
  
─  checking for empty or unneeded directories
#> 
  
─  building ‘opentripplanner_0.2.0.0000.tar.gz’
#> 
  
   
#> 
── Checking ──────────────────────────────────────────────────────────────────────────────────────────────────────── opentripplanner ──
#> Setting env vars:
#> ● _R_CHECK_CRAN_INCOMING_USE_ASPELL_: TRUE
#> ● _R_CHECK_CRAN_INCOMING_REMOTE_    : FALSE
#> ● _R_CHECK_CRAN_INCOMING_           : FALSE
#> ● _R_CHECK_FORCE_SUGGESTS_          : FALSE
#> ── R CMD check ─────────────────────────────────────────────────────────────────
#> * using log directory ‘/tmp/RtmpL5iiMU/opentripplanner.Rcheck’
#> * using R version 3.6.1 (2019-07-05)
#> * using platform: x86_64-pc-linux-gnu (64-bit)
#> * using session charset: UTF-8
#> * using options ‘--no-manual --as-cran’
#> * checking for file ‘opentripplanner/DESCRIPTION’ ... OK
#> * this is package ‘opentripplanner’ version ‘0.2.0.0000’
#> * package encoding: UTF-8
#> * checking package namespace information ... OK
#> * checking package dependencies ... OK
#> * checking if this is a source package ... OK
#> * checking if there is a namespace ... OK
#> * checking for executable files ... OK
#> * checking for hidden files and directories ... OK
#> * checking for portable file names ... OK
#> * checking for sufficient/correct file permissions ... OK
#> * checking serialization versions ... OK
#> * checking whether package ‘opentripplanner’ can be installed ... OK
#> * checking installed package size ... OK
#> * checking package directory ... OK
#> * checking for future file timestamps ... OK
#> * checking ‘build’ directory ... OK
#> * checking DESCRIPTION meta-information ... OK
#> * checking top-level files ... NOTE
#> Non-standard file/directory found at top level:
#>   ‘paper.bib’
#> * checking for left-over files ... OK
#> * checking index information ... OK
#> * checking package subdirectories ... OK
#> * checking R files for non-ASCII characters ... OK
#> * checking R files for syntax errors ... OK
#> * checking whether the package can be loaded ... OK
#> * checking whether the package can be loaded with stated dependencies ... OK
#> * checking whether the package can be unloaded cleanly ... OK
#> * checking whether the namespace can be loaded with stated dependencies ... OK
#> * checking whether the namespace can be unloaded cleanly ... OK
#> * checking loading without being on the library search path ... OK
#> * checking dependencies in R code ... OK
#> * checking S3 generic/method consistency ... OK
#> * checking replacement functions ... OK
#> * checking foreign function calls ... OK
#> * checking R code for possible problems ... OK
#> * checking Rd files ... OK
#> * checking Rd metadata ... OK
#> * checking Rd line widths ... NOTE
#> Rd file 'otp_dl_demo.Rd':
#>   \usage lines wider than 90 characters:
#>          url = "https://github.com/ropensci/opentripplanner/releases/download/0.1/isle-of-wight-demo.zip"
#> 
#> These lines will be truncated in the PDF manual.
#> * checking Rd cross-references ... OK
#> * checking for missing documentation entries ... OK
#> * checking for code/documentation mismatches ... OK
#> * checking Rd \usage sections ... OK
#> * checking Rd contents ... OK
#> * checking for unstated dependencies in examples ... OK
#> * checking installed files from ‘inst/doc’ ... OK
#> * checking files in ‘vignettes’ ... OK
#> * checking examples ... OK
#> * checking for unstated dependencies in ‘tests’ ... OK
#> * checking tests ...
#>   Running ‘testthat.R’
#>  ERROR
#> Running the tests in ‘tests/testthat.R’ failed.
#> Last 13 lines of output:
#>   
#>   [31m──[39m [31m1. Failure: can get lsoa points (@test_03_with_OTP.R#14) [39m [31m──────────────────────────────────────────────────────────────────────────[39m
#>   nrow(lsoa) == 89 isn't true.
#>   
#>   ══ testthat results  ══════════════════════════════════════════════════════════════════════════════════════════════════════════════════
#>   [ OK: 69 | SKIPPED: 19 | WARNINGS: 100 | FAILED: 1 ]
#>   1. Failure: can get lsoa points (@test_03_with_OTP.R#14) 
#>   
#>   Error: testthat unit tests failed
#>   In addition: Warning messages:
#>   1: In for (i in nsInfo$imports) { :
#>     closing unused connection 5 (<-localhost:11147)
#>   2: In for (i in nsInfo$imports) { :
#>     closing unused connection 4 (<-localhost:11147)
#>   Execution halted
#> * checking for unstated dependencies in vignettes ... OK
#> * checking package vignettes in ‘inst/doc’ ... OK
#> * checking re-building of vignette outputs ... OK
#> * DONE
#> 
#> Status: 1 ERROR, 2 NOTEs
#> See
#>   ‘/tmp/RtmpL5iiMU/opentripplanner.Rcheck/00check.log’
#> for details.
#> 
#> ── R CMD check results ───────────────────────── opentripplanner 0.2.0.0000 ────
#> Duration: 21.5s
#> 
#> ❯ checking tests ...
#>   See below...
#> 
#> ❯ checking top-level files ... NOTE
#>   Non-standard file/directory found at top level:
#>     ‘paper.bib’
#> 
#> ❯ checking Rd line widths ... NOTE
#>   Rd file 'otp_dl_demo.Rd':
#>     \usage lines wider than 90 characters:
#>            url = "https://github.com/ropensci/opentripplanner/releases/download/0.1/isle-of-wight-demo.zip"
#>   
#>   These lines will be truncated in the PDF manual.
#> 
#> ── Test failures ───────────────────────────────────────────────── testthat ────
#> 
#> > library(testthat)
#> > library(opentripplanner)
#> > 
#> > test_check("opentripplanner")
#> trying URL 'https://github.com/ropensci/opentripplanner/releases/download/0.1/isle-of-wight-demo.zip'
#> Content type 'application/octet-stream' length 8309996 bytes (7.9 MB)
#> ==================================================
#> downloaded 7.9 MB
#> 
#> trying URL 'https://repo1.maven.org/maven2/org/opentripplanner/otp/1.4.0/otp-1.4.0-shaded.jar'
#> Content type 'application/java-archive' length 61838786 bytes (59.0 MB)
#> ==================================================
#> downloaded 59.0 MB
#> 
#> [31m──[39m [31m1. Failure: can get lsoa points (@test_03_with_OTP.R#14) [39m [31m──────────────────────────────────────────────────────────────────────────[39m
#> nrow(lsoa) == 89 isn't true.
#> 
#> ══ testthat results  ══════════════════════════════════════════════════════════════════════════════════════════════════════════════════
#> [ OK: 69 | SKIPPED: 19 | WARNINGS: 100 | FAILED: 1 ]
#> 1. Failure: can get lsoa points (@test_03_with_OTP.R#14) 
#> 
#> Error: testthat unit tests failed
#> In addition: Warning messages:
#> 1: In for (i in nsInfo$imports) { :
#>   closing unused connection 5 (<-localhost:11147)
#> 2: In for (i in nsInfo$imports) { :
#>   closing unused connection 4 (<-localhost:11147)
#> Execution halted
#> 
#> 1 error ✖ | 0 warnings ✔ | 2 notes ✖
#> Error: R CMD check found ERRORs
```

<sup>Created on 2019-11-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mem48** [comment]
I'm running again but its fine on Travis and AppVeyor

**Comment by mem48** [comment]
```
-- R CMD check results ------------------------------------------------------------------------------- opentripplanner 0.2.0.0000 ----
Duration: 2m 59.5s

> checking for unstated dependencies in examples ... OK
   WARNING
  'qpdf' is needed for checks on size reduction of PDFs

> checking top-level files ... NOTE
  Non-standard file/directory found at top level:
    'paper.bib'

> checking Rd line widths ... NOTE
  Rd file 'otp_dl_demo.Rd':
    \usage lines wider than 90 characters:
           url = "https://github.com/ropensci/opentripplanner/releases/download/0.1/isle-of-wight-demo.zip"
  
  These lines will be truncated in the PDF manual.

0 errors √ | 1 warning x | 2 notes x
```

**Comment by Robinlovelace** [comment]
@mem48 can you give me push access to this new repo? I plan to push to a new branch and put in a PR.

**Comment by mem48** [comment]
@Robinlovelace done

**Comment by Robinlovelace** [comment]
Haha if you were 30 seconds faster! Just pushed to a new fork of it on my own account, good backup in case this new fork goes terribly wrong hehe.

**Comment by Robinlovelace** [comment]
That will fix these low hanging fruit:

```
Found the following (possibly) invalid URLs:
  URL: http://localhost:8080
    From: inst/doc/opentripplanner.html
    Status: Error
    Message: libcurl error code 7:
      	Failed to connect to localhost port 8080: Connection refused
  URL: http://localhost:8080?debug_layers=true
    From: inst/doc/advanced_features.html
    Status: Error
    Message: libcurl error code 7:
      	Failed to connect to localhost port 8080: Connection refused
  URL: https://cran.r-project.org/web/packages/sf/index.html
    From: README.md
    Status: 200
    Message: OK
    CRAN URL not in canonical form
  The canonical URL of the CRAN page for a package is 
    https://CRAN.R-project.org/package=pkgname
```

**Comment by Robinlovelace** [comment]
Update: no tests should be running that rely on OTP on my machine:

``` r
Sys.getenv("I_have_OTP")
#> [1] ""
```

<sup>Created on 2019-11-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by Robinlovelace** [comment]
I'll take a look at what's causing those more serious CRAN check failures.

**Comment by mem48** [comment]
I have and idea what might be the problem, could a firewall be blocking downloads? I test downlaoding all the files even if you don't have OTP. But if downloads are blocked that might cause some of these errors.

**Comment by Robinlovelace** [comment]
No that's not it...

**Comment by Robinlovelace** [comment]
One reason: 

``` r
lsoa <- sf::read_sf("https://github.com/ropensci/opentripplanner/releases/download/0.1/centroids.gpkg")
#> Warning in CPL_read_ogr(dsn, layer, query, as.character(options), quiet, : GDAL

```

<sup>Created on 2019-11-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by Robinlovelace** [comment]
This works, will put in a new PR:

``` r
download.file("https://github.com/ropensci/opentripplanner/releases/download/0.1/centroids.gpkg", "centroids.gpkg",  mode = "wb")
lsoa <- sf::read_sf("centroids.gpkg")
```

<sup>Created on 2019-11-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by Robinlovelace** [comment]
Good news: after #36 is merged I only get 2 notes:

``` r
devtools::check("~/wip/opentripplanner/")
#> Updating opentripplanner documentation
#> ...


#> * checking top-level files ... NOTE
#> Non-standard file/directory found at top level:
#>   ‘paper.bib’
#> Status: 2 NOTEs
#> See
#>   ‘/tmp/RtmpMk1rby/opentripplanner.Rcheck/00check.log’
#> for details.
#> ── R CMD check results ───────────────────────── opentripplanner 0.2.0.0000 ────
#> Duration: 20.6s
#> 
#> ❯ checking top-level files ... NOTE
#>   Non-standard file/directory found at top level:
#>     ‘paper.bib’
#> 
#> ❯ checking Rd line widths ... NOTE
#>   Rd file 'otp_dl_demo.Rd':
#>     \usage lines wider than 90 characters:
#>            url = "https://github.com/ropensci/opentripplanner/releases/download/0.1/isle-of-wight-demo.zip"
#>   
#>   These lines will be truncated in the PDF manual.
#> 
#> 0 errors ✔ | 0 warnings ✔ | 2 notes ✖
```

<sup>Created on 2019-11-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mem48** [comment]
THanks for rapit help, I'm going to run the full set of tests locally and then resubmit to CRAN

**Comment by Robinlovelace** [comment]
OK great. I'm about to push a commit to close this. Zero notes now!

**Comment by Robinlovelace** [comment]
:rocket: :dancer: 

```
── R CMD check results ───────────────────────── opentripplanner 0.2.0.0001 ────
Duration: 20.2s

0 errors ✔ | 0 warnings ✔ | 0 notes ✔

R CMD check succeeded

```

**Comment by mem48** [comment]
@Robinlovelace this is still happening

package opentripplanner_0.2.0.2.tar.gz does not pass the incoming checks automatically, please see the following pre-tests:
Windows: <https://win-builder.r-project.org/incoming_pretest/opentripplanner_0.2.0.2_20191130_231531/Windows/00check.log>
Status: 1 ERROR, 1 NOTE
Debian: <https://win-builder.r-project.org/incoming_pretest/opentripplanner_0.2.0.2_20191130_231531/Debian/00check.log>
Status: 1 ERROR, 1 NOTE

The errors are different for Windows and Debian. On windows it says knitr is unavailable despite being listed in the DESCRIPTION file. On Debian some of the tests are failing. 

All of the failing tests relate to the same function ` otp_clean_input` (https://github.com/ropensci/opentripplanner/blob/master/R/otp-plan.R#L303)

I can’t work out how to fix the problems because. The problem seems to be unique to R 4.0

If I run checks on the whole package 

`devtools::check()`

I see the same errors at the CRAN checks e.g.

```
-- 1. Failure: test otp_clean_input (@test_01_
`otp_clean_input(c(181, 2), "foo")` threw an error with unexpected message.
Expected match: "is not <= 180"
Actual message: "foo is not in a valid format"
Backtrace:
1. testthat::expect_error(otp_clean_input(c(181, 2), "foo"), regexp = "is not <= 180")
6. opentripplanner:::otp_clean_input(c(181, 2), "foo")
```

But if I run,

`devtools::test()`

I get no errors. This is also true if I run tests one at a time, for example.

```
opentripplanner:::otp_clean_input(c(181, 2), "foo")
Error in opentripplanner:::otp_clean_input(c(181, 2), "foo") : 
  Assertion on 'imp[, 1]' failed: Element 1 is not <= 180.
```

This is the correct error message for this input, and example of the function working correctly is:

```
opentripplanner:::otp_clean_input(c(179, 2), "foo")
     [,1] [,2]
[1,]    2  179
```

**Comment by layik** [comment]
Hi @mem48. I have no idea if this would be the reason but on a clean build from me, I have this note:
UPDATE: I have an idea and this was not it but still valid see next comment.
```
Missing or unexported object: ‘utils::askYesNo’
```
<details><summary> sessionInfo() </summary>
``` r
#> R version 3.4.4 (2018-03-15)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Ubuntu 18.04.3 LTS
#> 
#> Matrix products: default
#> BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
#> LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1
#> 
#> locale:
#>  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
#>  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
#>  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
#>  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
#>  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#> [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> loaded via a namespace (and not attached):
#>  [1] compiler_3.4.4  magrittr_1.5    tools_3.4.4     htmltools_0.4.0
#>  [5] yaml_2.2.0      Rcpp_1.0.3      stringi_1.4.3   rmarkdown_1.18 
#>  [9] highr_0.8       knitr_1.26      stringr_1.4.0   xfun_0.11      
#> [13] digest_0.6.23   rlang_0.4.2     evaluate_0.14
```

<sup>Created on 2019-12-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
</details>
I will try to contribute!

**Comment by layik** [comment]
Right, regardless of the fail on those platforms `askYesNo` is [not available](https://www.rdocumentation.org/packages/utils/versions/3.4.3) at least on `utils` up to utils version `3.5`. Need to be aware of this. Happy to send a PR for that and also avoid `interactive` session.

I think we should deal with this old version issue by restricting the `utils` package version itself like done in [stats19](https://github.com/ropensci/stats19/blob/master/DESCRIPTION#L39) or 

`otp-setup` file line 277
```
&&
        # TODO check interactive
        # https://www.rdocumentation.org/packages/utils/versions/3.4.3
        # askYesNo is not available in utils upto 3.5
        exists('askYesNo', where='package:utils', mode='function')

**Comment by layik** [comment]
And to confirm, upon upgrading to 3.6 (which means losing packages) the note disappears. So checking through the error on Linux, I _**think**_ the culprit is this:
https://github.com/ropensci/opentripplanner/blob/master/R/otp-plan.R#L302

not exported. Sending a PR.

**Comment by layik** [comment]
#39

**Comment by Robinlovelace** [comment]
It's simply that the test fails, not an OS specific issue as Layik says.

**Comment by mem48** [comment]
@Robinlovelace but the vignettes build on Linux but not on Windows so ther must be an OS aspect to it.

**Comment by Robinlovelace** [comment]
The test is wrong. You ask it to expect an error but there is not error generated:

``` r
opentripplanner:::otp_clean_input(c(179, 2), "foo")
#>      [,1] [,2]
#> [1,]    2  179
```

<sup>Created on 2019-12-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I suggest closing this issue and opening a new one on the tests. If the Windoes vignettes are failing that's another issue.

**Comment by Robinlovelace** [comment]
Update: I think the test is wrong, not 100% sure what it's trying to test.

**Comment by Robinlovelace** [comment]
Update: OK the test passes on my machine:

``` r
testthat::expect_error(opentripplanner:::otp_clean_input(c(181, 
    2), "foo"), regexp = "is not <= 180")
```

<sup>Created on 2019-12-02 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by layik** [comment]
Yes, whatever it is, it is to do with the tests. Lets use the same thread to fix the tests :)

**Comment by mem48** [comment]
@layik your PR is still failing on R 4.0 :(

**Comment by layik** [comment]
Can I please see the log? Is it the above issue with `askYesNo`?

**Comment by mem48** [comment]
@layik 

```
Updating opentripplanner documentation
Writing NAMESPACE
Loading opentripplanner
Writing NAMESPACE
-- Building ------------------------------------ opentripplanner --
Setting env vars:
* CFLAGS    : -Wall -pedantic
* CXXFLAGS  : -Wall -pedantic
* CXX11FLAGS: -Wall -pedantic
-------------------------------------------------------------------
v  checking for file 'C:\Users\Malcolm\Documents\GitHub\ropensci\opentripplanner/DESCRIPTION' (551ms)
-  preparing 'opentripplanner': (2.5s)
v  checking DESCRIPTION meta-information ... 
-  installing the package to build vignettes
v  creating vignettes (7.9s)
-  checking for LF line-endings in source and make files and shell scripts (736ms)
-  checking for empty or unneeded directories
-  building 'opentripplanner_0.2.0.2.tar.gz'
   
-- Checking ------------------------------------ opentripplanner --
Setting env vars:
* _R_CHECK_CRAN_INCOMING_REMOTE_: FALSE
* _R_CHECK_CRAN_INCOMING_       : FALSE
* _R_CHECK_FORCE_SUGGESTS_      : FALSE
-- R CMD check -----------------------------------------------------------------
* using log directory 'C:/Users/Malcolm/Documents/GitHub/ropensci/opentripplanner.Rcheck'
* using R Under development (unstable) (2019-11-29 r77486)
* using platform: x86_64-w64-mingw32 (64-bit)
* using session charset: ISO8859-1
* using options '--no-manual --as-cran'
* checking for file 'opentripplanner/DESCRIPTION' ... OK
* this is package 'opentripplanner' version '0.2.0.2'
* package encoding: UTF-8
* checking package namespace information ... OK
* checking package dependencies ... NOTE
Package suggested but not available for checking: 'rmarkdown'
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for .dll and .exe files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking serialization versions ... OK
* checking whether package 'opentripplanner' can be installed ... OK
* checking package directory ... OK
* checking for future file timestamps ... OK
* checking 'build' directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd line widths ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking installed files from 'inst/doc' ... OK
* checking files in 'vignettes' ... OK
* checking examples ... OK
* checking for unstated dependencies in 'tests' ... OK
* checking tests ...
  Running 'testthat.R' [48s]
 ERROR
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
  == testthat results  ==============================================
  [ OK: 47 | SKIPPED: 19 | WARNINGS: 2 | FAILED: 14 ]
  1. Failure: test otp_clean_input (@test_01_internal_funcs.R#18) 
  2. Failure: test otp_clean_input (@test_01_internal_funcs.R#19) 
  3. Failure: test otp_clean_input (@test_01_internal_funcs.R#20) 
  4. Failure: test otp_clean_input (@test_01_internal_funcs.R#21) 
  5. Error: test otp_clean_input (@test_01_internal_funcs.R#25) 
  6. Failure: otp_plan input validation (@test_02_without_OTP.R#80) 
  7. Failure: otp_plan input validation (@test_02_without_OTP.R#83) 
  8. Failure: otp_plan input validation (@test_02_without_OTP.R#89) 
  9. Failure: otp_plan input validation (@test_02_without_OTP.R#95) 
  1. ...
  
  Error: testthat unit tests failed
  Execution halted
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in 'inst/doc' ... OK
* checking re-building of vignette outputs ... OK
* checking for non-standard things in the check directory ... OK
* checking for detritus in the temp directory ... OK
* DONE

Status: 1 ERROR, 1 NOTE
See
  'C:/Users/Malcolm/Documents/GitHub/ropensci/opentripplanner.Rcheck/00check.log'
for details.

-- R CMD check results ---------------------------- opentripplanner 0.2.0.2 ----
Duration: 1m 28.7s

> checking tests ...
  See below...

> checking package dependencies ... NOTE
  Package suggested but not available for checking: 'rmarkdown'

-- Test failures ------------------------------------------------- testthat ----

> library(testthat)
> library(opentripplanner)
> 
> test_check("opentripplanner")
-- 1. Failure: test otp_clean_input (@test_01_internal_funcs.R#18) 
`otp_clean_input(c(181, 2), "foo")` threw an error with unexpected message.
Expected match: "is not <= 180"
Actual message: "foo is not in a valid format"
Backtrace:
 1. testthat::expect_error(otp_clean_input(c(181, 2), "foo"), "is not <= 180")
 6. opentripplanner:::otp_clean_input(c(181, 2), "foo")

-- 2. Failure: test otp_clean_input (@test_01_internal_funcs.R#19) 
`otp_clean_input(c(-181, 2), "foo")` threw an error with unexpected message.
Expected match: "is not >= -180"
Actual message: "foo is not in a valid format"
Backtrace:
 1. testthat::expect_error(otp_clean_input(c(-181, 2), "foo"), "is not >= -180")
 6. opentripplanner:::otp_clean_input(c(-181, 2), "foo")

-- 3. Failure: test otp_clean_input (@test_01_internal_funcs.R#20) 
`otp_clean_input(c(0, 91), "foo")` threw an error with unexpected message.
Expected match: "is not <= 90"
Actual message: "foo is not in a valid format"
Backtrace:
 1. testthat::expect_error(otp_clean_input(c(0, 91), "foo"), "is not <= 90")
 6. opentripplanner:::otp_clean_input(c(0, 91), "foo")

-- 4. Failure: test otp_clean_input (@test_01_internal_funcs.R#21) 
`otp_clean_input(c(0, -91), "foo")` threw an error with unexpected message.
Expected match: "is not >= -90"
Actual message: "foo is not in a valid format"
Backtrace:
 1. testthat::expect_error(otp_clean_input(c(0, -91), "foo"), "is not >= -90")
 6. opentripplanner:::otp_clean_input(c(0, -91), "foo")

-- 5. Error: test otp_clean_input (@test_01_internal_funcs.R#25)  -
foo is not in a valid format
Backtrace:
 1. opentripplanner:::otp_clean_input(c(1, 2), "foo")

trying URL 'https://github.com/ropensci/opentripplanner/releases/download/0.1/isle-of-wight-demo.zip'
Content type 'application/octet-stream' length 8309996 bytes (7.9 MB)
==================================================
downloaded 7.9 MB

trying URL 'https://repo1.maven.org/maven2/org/opentripplanner/otp/1.4.0/otp-1.4.0-shaded.jar'
Content type 'application/java-archive' length 61838786 bytes (59.0 MB)
==================================================
downloaded 59.0 MB

-- 6. Failure: otp_plan input validation (@test_02_without_OTP.R#80
`otp_plan(otpcon, fromPlace = c(1.23, 1.23))` threw an error with unexpected message.
Expected match: "toPlace is not in a valid format"
Actual message: "fromPlace is not in a valid format"
Backtrace:
 1. testthat::expect_error(...)
 6. opentripplanner::otp_plan(otpcon, fromPlace = c(1.23, 1.23))
 7. opentripplanner:::otp_clean_input(fromPlace, "fromPlace")

-- 7. Failure: otp_plan input validation (@test_02_without_OTP.R#83
`otp_plan(...)` threw an error with unexpected message.
Expected match: "Failed to connect to localhost port 8080: Connection refused"
Actual message: "fromPlace is not in a valid format"
Backtrace:
 1. testthat::expect_error(...)
 6. opentripplanner::otp_plan(...)
 7. opentripplanner:::otp_clean_input(fromPlace, "fromPlace")

-- 8. Failure: otp_plan input validation (@test_02_without_OTP.R#89
`otp_plan(...)` threw an error with unexpected message.
Expected match: "Number of fromPlaces and toPlaces do not match"
Actual message: "fromPlace is not in a valid format"
Backtrace:
 1. testthat::expect_error(...)
 6. opentripplanner::otp_plan(...)
 7. opentripplanner:::otp_clean_input(fromPlace, "fromPlace")

-- 9. Failure: otp_plan input validation (@test_02_without_OTP.R#95
`otp_plan(...)` threw an error with unexpected message.
Expected match: "Failed to connect to localhost port 8080: Connection refused"
Actual message: "fromPlace is not in Error: R CMD check found ERRORs
a valid format"
Backtrace:
 1. testthat::expect_error(...)
 6. opentripplanner::otp_plan(...)
 7. opentripplanner:::otp_clean_input(fromPlace, "fromPlace")

-- 10. Failure: otp_plan input validation (@test_02_without_OTP.R#1
`otp_plan(...)` threw an error with unexpected message.
Expected match: "The length of toID and toPlace are not the same"
Actual message: "fromPlace is not in a valid format"
Backtrace:
 1. testthat::expect_error(...)
 6. opentripplanner::otp_plan(...)
 7. opentripplanner:::otp_clean_input(fromPlace, "fromPlace")

-- 11. Failure: otp_plan input validation (@test_02_without_OTP.R#1
`otp_plan(...)` threw an error with unexpected message.
Expected match: "The length of fromID and fromPlace are not the same"
Actual message: "fromPlace is not in a valid format"
Backtrace:
 1. testthat::expect_error(...)
 6. opentripplanner::otp_plan(...)
 7. opentripplanner:::otp_clean_input(fromPlace, "fromPlace")

-- 12. Failure: otp_plan input validation (@test_02_without_OTP.R#1
`otp_plan(...)` threw an error with unexpected message.
Expected match: "Failed to connect to localhost port 8080: Connection refused"
Actual message: "fromPlace is not in a valid format"
Backtrace:
 1. testthat::expect_error(...)
 6. opentripplanner::otp_plan(...)
 7. opentripplanner:::otp_clean_input(fromPlace, "fromPlace")

-- 13. Failure: otp_plan input validation (@test_02_without_OTP.R#1
`otp_plan(...)` threw an error with unexpected message.
Expected match: "Failed to connect to localhost port 8080: Connection refused"
Actual message: "fromPlace is not in a valid format"
Backtrace:
 1. testthat::expect_error(...)
 6. opentripplanner::otp_plan(...)
 7. opentripplanner:::otp_clean_input(fromPlace, "fromPlace")

trying URL 'https://github.com/ropensci/opentripplanner/releases/download/0.1/centroids.gpkg'
Content type 'application/octet-stream' length 126976 bytes (124 KB)
==================================================
downloaded 124 KB

-- 14. Error: basic routing (@test_04_with_public_API.R#10)  ------
fromPlace is not in a valid format
Backtrace:
 1. opentripplanner::otp_plan(...)
 2. opentripplanner:::otp_clean_input(fromPlace, "fromPlace")

== testthat results  ==============================================
[ OK: 47 | SKIPPED: 19 | WARNINGS: 2 | FAILED: 14 ]
1. Failure: test otp_clean_input (@test_01_internal_funcs.R#18) 
2. Failure: test otp_clean_input (@test_01_internal_funcs.R#19) 
3. Failure: test otp_clean_input (@test_01_internal_funcs.R#20) 
4. Failure: test otp_clean_input (@test_01_internal_funcs.R#21) 
5. Error: test otp_clean_input (@test_01_internal_funcs.R#25) 
6. Failure: otp_plan input validation (@test_02_without_OTP.R#80) 
7. Failure: otp_plan input validation (@test_02_without_OTP.R#83) 
8. Failure: otp_plan input validation (@test_02_without_OTP.R#89) 
9. Failure: otp_plan input validation (@test_02_without_OTP.R#95) 
1. ...

Error: testthat unit tests failed
Execution halted

1 error x | 0 warnings v | 1 note x
Execution halted

Exited with status 1.

WARNING: Rtools is required to build R packages but no version of Rtools compatible with the currently running version of R was found. Note that the following incompatible version(s) of Rtools were found:

  - Rtools 3.5 (installed at C:\Rtools)

Please download and install the appropriate version of Rtools before proceeding:

https://cran.rstudio.com/bin/windows/Rtools/
```

**Comment by Robinlovelace** [comment]
This seems like a pretty clear indication of what's going on. Not sure why it's giving that message though...

```
Expected match: "Failed to connect to localhost port 8080: Connection refused"
Actual message: "fromPlace is not in a valid format"
```

**Comment by Robinlovelace** [comment]
Maybe it's a class issue, is `fromPlace` supposed to be a particular class (e.g. numeric)?

**Comment by layik** [comment]
I think what is happening is the test case is trying to match provided error message with actual error message from function.

**Comment by mem48** [comment]
@layik your PR will work, but because you have disabled the error checking. The question is why are the wrong error messages being returned?

What these test are doing is checking that error handling is working correctly. Here is a simplified example

```
func <- function(x){
  if(class(x) == "numeric"){
    return(x)
  }else{
    error(x," is not numeric")
  }
}
func(123)
#> [1] 123
```

If the test was returning the message "123 is not numeric" there is something going very wrong.

**Comment by layik** [comment]
I see, my apologies, just closed my PR. In that case, I need to check.

**Comment by layik** [comment]
https://stackoverflow.com/a/30063199/2332101

**Comment by layik** [comment]
If I know how to send a build, I would send one with adding `fixed = TRUE` as stated in that SO answer. But wont trouble you with another PR. I think I should stop here if this is not the solution.

**Comment by mem48** [comment]
@layik  tired adding the `fixed = TRUE` but didn't make any difference

**Comment by mem48** [comment]
Fixed, turn of the matrix was gaining the class "array" which was breaking the test `all(class(imp) == "matrix")`

**Comment by Robinlovelace** [comment]
Guessed it was to do with the class of the inputs. Good fix @mem48 !

**Comment by layik** [comment]
Hopefully submission to CRAN next?
Is there a ticket to submit to CRAN? I think there was a STATS19 ticket outlining the checklist.
